### DISCUSSION TITLE
How to enable float64 globally?

### DISCUSSION THREAD

**Discussion by zgbkdlm** [google/flax#3044] [Q&A] [answered]
Hi,

Can I ask how to enable float64 globally? 

I tried to use `jax.config.update("jax_enable_x64", True)`, but this does not work, and the resulting neural network parameters are still float32. I also tried to add `dtype` arguments in the model creation, but it does not work either.

```python
class NN(nn.Module):

    dtype = jax.numpy.float64

    @nn.compact
    def __call__(self, x):
        x = nn.Dense(features=10, dtype=jax.numpy.float64)(x)
        x = nn.gelu(x)
        x = nn.Dense(features=5, dtype=jax.numpy.float64)(x)
        x = nn.gelu(x)
        x = nn.Dense(features=1, dtype=jax.numpy.float64)(x)
        return x


model = NN()
params = model.init(key, dataset.xs)
print(params)
```

**Comment by soraros** [comment, marked as answer]
Flax defaults parameter dtype to `flat32` which can be changed in your case by setting [`param_dtype`](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.Dense.html#flax.linen.Dense) to `np.float64`.

**Reply by jiaxi98** [reply]
Yeah, this method works, but is that possible to do this globally? Otherwise for a UNet implementation, I have to specify the param_dtype for every Conv blocks.
